packages/medusa-react/src/hooks/store/payment-collections/mutations.ts

```diff
@@ -0,0 +1,90 @@
+import { useMutation, UseMutationOptions, useQueryClient } from "react-query"
+import { Response } from "@medusajs/medusa-js"
+
+import {
+  StorePaymentCollectionRes,
+  StoreManagePaymentCollectionSessionRequest,
+  StoreRefreshPaymentCollectionSessionRequest,
+} from "@medusajs/medusa"
+
+import { buildOptions } from "../../utils/buildOptions"
+import { useMedusa } from "../../../contexts"
+import { paymentCollectionQueryKeys } from "."
+
+export const useManagePaymentSessions = (
+  id: string,
+  options?: UseMutationOptions<
+    Response<StorePaymentCollectionRes>,
+    Error,
+    StoreManagePaymentCollectionSessionRequest
+  >
+) => {
+  const { client } = useMedusa()
+  const queryClient = useQueryClient()
+
+  return useMutation(
+    (payload: StoreManagePaymentCollectionSessionRequest) =>
+      client.paymentCollections.manageSessions(id, payload),
+    buildOptions(
+      queryClient,
+      [
+        paymentCollectionQueryKeys.lists(),
+        paymentCollectionQueryKeys.detail(id),
+      ],
+      options
+    )
+  )
+}
+
+export const useAuthorize = (
```

Comment:
suggestion(non-blocking): Rename to remove ambiguity
```suggestion
export const useAuthorizePayment[Session | Collection] = (
```

Reply:
Not entirely sure, if we should use session or collection here. Collection is primarily a backend concept, which is why session might make sense. What do you think?

Reply:
The payCol authorization will try to authorize all payment sessions that are not yet authorized.
Do you think we should have an authorizeSession method that authorizes session by session?